server/implementation/src/main/java/io/smallrye/graphql/bootstrap/Bootstrap.java

```diff
@@ -183,7 +186,30 @@ private void generateGraphQLSchema() {
         Map<String, Jsonb> overrides = eventEmitter.fireOverrideJsonbConfig();
         JsonInputRegistry.override(overrides);
 
-        this.graphQLSchema = schemaBuilder.build();
+        if (Config.get().isFederationEnabled()) {
+            log.enableFederation();
+            GraphQLSchema rawSchema = schemaBuilder.build();
+            this.graphQLSchema = Federation.transform(rawSchema)
+                    .fetchEntities(new FederationDataFetcher(rawSchema.getQueryType(), rawSchema.getCodeRegistry()))
+                    .resolveEntityType(fetchEntityType())
+                    .build();
+        } else {
+            this.graphQLSchema = schemaBuilder.build();
+        }
+    }
+
+    private TypeResolver fetchEntityType() {
+        return env -> {
+            Object src = env.getObject();
+            if (src == null) {
+                return null;
+            }
+            GraphQLObjectType result = env.getSchema().getObjectType(src.getClass().getSimpleName()); // TODO respect @Name, etc.
```

Comment:
To allow the use of @Name

```
Class<?> clazz = src.getClass();
String typeName = clazz.getSimpleName();
Name annotation = clazz.getAnnotation(Name.class);
if (annotation!=null) {
  typeName = annotation.value();
}
GraphQLObjectType result = env.getSchema().getObjectType(typeName);
```

Reply:
I think we'll have to use Jandex to find the annotation.

Reply:
@phillip-kruger : can you give us a hint on how to resolve this best? I already searched for some helper method.

Reply:
Shouldn't something like this work
`
ScanningContext.getIndex().getClassByName(typeName)
...
`

Reply:
The `Bootstrap` class is in `server/implementation` which doesn't have a dependency on `common/schema-builder`, so I can't access the `ScanningContext`. There's also no dependency on Jandex, so I think we should go for the minimalistic solution  that @johgoe has suggested... for now. If a better solution comes up, we can do that later.

Reply:
Done. Thanks, all!